File: SEP3_T2/grpc-springboot/src/main/java/sep3/dto/post/UpdatePostDTO.java
package sep3.dto.post;

/**
 * Data transfer object representing a post to be updated, including title, content,
 * category ID, and post ID.
 */
public class UpdatePostDTO {
    private String title;
    private String content;
    private int postId;
    private int categoryId;

    /**
     * Constructs a new UpdatePostDTO with the specified title, content, category ID, and post ID.
     *
     * @param title      The new title for the post.
     * @param content    The new content/body of the post.
     * @param categoryId The ID of the category to which the post belongs.
     * @param postId     The unique ID of the post to be updated.
     */
    public UpdatePostDTO(String title, String content, int categoryId, int postId) {
        this.title = title;
        this.content = content;
        this.categoryId = categoryId;
        this.postId = postId;
    }

    /**
     * Retrieves the new title for the post.
     *
     * @return The new title of the post.
     */
    public String getTitle() {
        return title;
    }

    /**
     * Retrieves the new content/body of the post.
     *
     * @return The new content of the post.
     */
    public String getContent() {
        return content;
    }

    /**
     * Retrieves the category ID of the post.
     *
     * @return The category ID.
     */
    public int getCategoryId() {
        return categoryId;
    }

    /**
     * Retrieves the unique ID of the post to be updated.
     *
     * @return The post ID.
     */
    public int getPostId() {
        return postId;
    }
}

File: SEP3_T1/Shared/Domain/DTOs/Post/UpdatePostDTO.cs
namespace DTOs.User.PostDTOs;

public class UpdatePostDTO
{
    public int PostId { get; set; }
    public string Title { get; set; }
    public string Body { get; set; }
    public int AccountId { get; set; }
    public int CategoryId { get; set; }
    
    public UpdatePostDTO(int postId, string title, string body, int accountId, int categoryId)
    {
        PostId = postId;
        Title = title;
        Body = body;
        AccountId = accountId;
        CategoryId = categoryId;
    }
}